src/App.tsx

```diff
@@ -1,61 +1,121 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { ITodoListItem } from './interface/ITodoListItem';
+import { IUser } from './interface/IUser';
+import { TodoList } from './components/TodoList';
+import { useRef, useState } from 'react';
+
+function getUserById(userId: number): IUser | null {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+export const todos: ITodoListItem[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const formRef = useRef<HTMLFormElement>(null);
+  const [todosList, setTodosList] = useState<ITodoListItem[]>(todos);
+  const [titleValue, setTitleValue] = useState<string>('');
+  const [selectedUserId, setSelectedUserId] = useState<number>(0);
+  const [hasTouchedTitleInput, setHasTouchedTitleInput] =
+    useState<boolean>(false);
+  const [hasTouchedUserInput, setHasTouchedUserInput] =
+    useState<boolean>(false);
+  const hasValidatedTitle = !titleValue && hasTouchedTitleInput;
+  const hasValidatedUser = selectedUserId === 0 && hasTouchedUserInput;
+  const findMaxId = (): number => {
+    return todosList.reduce((accumulator, current) =>
+      accumulator.id > current.id ? accumulator : current,
+    ).id;
```

Comment:
The `reduce` method is used incorrectly here. It should return a number, but currently, it returns an object. You should compare `accumulator` and `current` based on their `id` values and return the maximum `id` instead of the entire object.